app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/TrackerBlockingVpnService.kt

```diff
@@ -222,8 +224,18 @@ class TrackerBlockingVpnService : VpnService(), CoroutineScope by MainScope() {
     private suspend fun startVpn() = withContext(Dispatchers.IO) {
         logcat { "VPN log: Starting VPN" }
 
-        // We need to rethink how to log this state. This will likely change.
-        vpnServiceStateStatsDao.insert(VpnServiceStateStats(state = ENABLING))
+        synchronized(startVpnLock) {
+            val currStateStats = vpnServiceStateStatsDao.getLastStateStats()
+            if (currStateStats?.state == ENABLING) {
+                // Sometimes onStartCommand gets called twice - this is a safety rail against that
+                logcat(LogPriority.WARN) { "VPN is already being started, abort" }
+                return@withContext
```

Comment:
Can we add a pixel here to get a sense how often this was happening?
As this is surely source for breakage when it happens, it'd be good to know how often it actually happened (?)